crates/cext/src/transpiler/passes/commutative_cancellation.rs

```diff
@@ -0,0 +1,124 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::commutation_checker::get_standard_commutation_checker;
+use qiskit_transpiler::passes::cancel_commutations;
+use qiskit_transpiler::target::Target;
+
+/// @ingroup QkTranspilerPasses
+/// Run the CommutativeCancellation transpiler pass on a circuit.
```

Comment:
Regarding the term CommutativeCancellation: We should think in general how much we want to refer to the Python documentation in the future. Is Python going to remain the main point for docs or are we going to have more generic documentation that both C and Python point to? As long as we point to Python it'd be really nice if we had hyperrefs though... but nothing for this PR ofc 🙂